libs/obs/src/CSimpleMap.cpp

```diff
@@ -233,6 +233,26 @@ void CSimpleMap::insert(
 	insert(CPose3DPDF::Ptr(CPose3DPDF::createFrom2D(*in_posePDF)), in_SF);
 }
 
+void CSimpleMap::insertToPos(
+	int index, const CPose3DPDF::Ptr& in_posePDF,
+	const CSensoryFrame::Ptr& in_SF)
+{
+	int size = m_posesObsPairs.size();
```

Comment:
@LisGein : I know this is "unlikely" to happen in this case but... `int` typically has 32bit size on 32bit builds, while `.size()` returns `size_t` (or `std::size_t`) which is typically 64bit, so comparing them makes me nervous...  (in some extreme cases, like when comparing against `std::string::npos`, I have seen these errors leading to crashes!!).

The best solution would be use use `size_t` for the type of the incoming `index` parameter. 
Also, please, add a brief Doxygen documentation to the declaration of this method in the .h

Also: should the ">" in `(index > size)` be a ">=" ?? (Not sure, just asking... it depends on the API of `std::deque::insert()`)

(You can just add a new commit to your fork and it'll automatically added to this PR)